### PR TITLE
Refactor screenshot capture module for cohesion and extensibility

### PR DESCRIPTION
**Pull request by JezaChen**
### Summary

- **Decouple screenshot logic from `Desktop`**: Move screenshot-related state and functions out of `Desktop` (service.py) into the `screenshot` module, eliminating unused delegation methods and reducing `Desktop`'s responsibilities.
- **Simplify `capture()` parameters**: Remove unnecessary callable/injection parameters (`crop_screenshot`, `get_monitors_rect`, `dxcam_module`, `mss_module`, `camera_cache`) that were passed through solely for testability but made the API confusing and brittle. Each backend now accesses what it needs directly.
- **Introduce `ScreenshotBackend` class hierarchy**: Replace the flat `capture_with_*` functions and manual if/elif dispatch with an OOP design using `__init_subclass__` for automatic backend registration. Adding a new backend now only requires defining a single class with `name` and `priority`.
- **Add comprehensive unit tests**: New `test_screenshot_capture.py` with 38 test cases covering the `capture()` API, each backend class, the auto-registration mechanism, and edge cases like fallback chains and error recovery.

### Motivation

The `Desktop` class had accumulated several screenshot-related methods (`_capture_with_dxcam`, `_capture_with_pillow`, `_get_dxcam_camera`, `_resolve_dxcam_region`, `_get_screenshot_backend`, `_crop_screenshot`, `_build_crop_box`) that were thin wrappers delegating to `screenshot.py` functions. These wrappers added no value — they simply forwarded calls — yet inflated `Desktop`'s surface area and obscured where the real logic lived.

The `capture()` function signature had grown to accept 7 parameters, most of which were dependency-injection hooks introduced purely for unit testing:

```python
# Before: confusing signature with test-only parameters
def capture(
    capture_rect,
    crop_screenshot: Callable,      # always Desktop._crop_screenshot
    get_monitors_rect: Callable,    # always uia.GetMonitorsRect
    camera_cache: dict,             # always Desktop._dxcam_cameras
    backend: str | None = None,
    dxcam_module=None,              # always the dxcam module
    mss_module=None,                # always the mss module
) -> tuple[Image.Image, str]:
```

This created several problems:
- **Unnecessary indirection**: Every call site passed the same objects. If the actual argument never varies, the function should just use it directly.
- **Backend-specific parameters leaking into the generic API**: `camera_cache` and `dxcam_module` only matter for the dxcam backend; passing them through the generic `capture()` is misleading — callers might expect them to affect other backends.
- **Module-level globals introduced for testability**: `_DXCAM_CAMERA_CACHE` was a module global dict that existed primarily so tests could reset it, making the code harder to follow.

### What changed

**`screenshot.py`** — Major restructure:
- Introduced `_ScreenshotBackend` base class with `__init_subclass__` auto-registration into `registry`.
- Three backend subclasses: `_DxcamBackend` (priority=10), `_MssBackend` (priority=20), `_PillowBackend` (priority=100).
- Each backend encapsulates its own availability check (`is_available`) and capture logic (`capture`).
- `_DxcamBackend` owns its camera cache as an instance attribute and its monitor region resolution as a static method.
- `capture()` reduced to 2 parameters: `capture(capture_rect, backend=None)`.
- `get_screenshot_backend()` validates against the dynamic registry instead of a hardcoded set.
- Removed: `capture_with_dxcam()`, `capture_with_mss()`, `capture_with_pillow()`, `get_dxcam_camera()`, `_DXCAM_CAMERA_CACHE`, `resolve_dxcam_region()`, `_auto_backend_chain()`.

**`service.py`** — Simplified:
- Removed `Desktop._dxcam_cameras`, `_get_screenshot_backend`, `_resolve_dxcam_region`, `_get_dxcam_camera`, `_capture_with_dxcam`, `_capture_with_pillow`, `_crop_screenshot`, `_build_crop_box`.
- Removed module-level `dxcam = screenshot_capture.dxcam` / `mss = screenshot_capture.mss` aliases.
- `Desktop.get_screenshot()` now simply calls `screenshot_capture.capture(capture_rect)`.

**`test_snapshot_display_filter.py`** — Updated mock targets:
- All `patch("windows_mcp.desktop.service.dxcam", ...)` / `service.mss` references replaced with `screenshot.*` targets.
- `_DXCAM_CAMERA_CACHE` mock replaced with `_backend_instances` reset.

**`test_screenshot_capture.py`** — New file, 38 test cases:
- `TestBackendRegistry`: auto-registration, priority ordering, incomplete subclass handling.
- `TestGetScreenshotBackend`: valid/invalid env vars, defaults, case insensitivity.
- `TestDxcamBackend`: availability checks, region resolution (exact match, sub-region coordinates, cross-monitor), camera caching, capture success/failure.
- `TestMssBackend`: availability, monitor dict construction, full-screen path.
- `TestPillowBackend`: always-available, error fallback with/without capture rect.
- `TestCapture`: explicit backend, unknown backend error, auto chain fallback, exception recovery, safety fallback, image content verification.

### Test plan

- [x] `ruff check . && ruff format --check .` — lint passes
- [x] `pytest tests/test_screenshot_capture.py -v` — 38 new tests pass
- [x] `pytest tests/test_snapshot_display_filter.py -v` — 18 existing tests pass (no regressions)

### CODE REVIEW

File: src/windows_mcp/desktop/screenshot.py [hunk 1]
```diff
@@ -30,154 +53,209 @@ def get_screenshot_backend() -> str:
...
+        except (OSError, RuntimeError, ValueError):
+            if capture_rect is not None:
+                logger.warning(
+                    "Failed to capture selected region directly, "
+                    "falling back to virtual screen crop"
+                )
+                return _crop_screenshot(ImageGrab.grab(all_screens=True), capture_rect)
+            logger.warning("Failed to capture virtual screen, using primary screen")
+            screenshot = ImageGrab.grab()
+        return _crop_screenshot(screenshot, capture_rect)
```
[the remark this answers is not in the document]
Comment (JezaChen) [reply src/windows_mcp/desktop/screenshot.py:181]: The double-crop issue in `_PillowBackend` and `_MssBackend` is a pre-existing bug (late discovery) — it existed in the original `capture_with_pillow` and `capture_with_mss` functions before this refactoring. Our refactoring moved the logic into class methods as-is without introducing or fixing it.

Root cause: When capture_rect is provided, both `ImageGrab.grab(bbox=...)` and `mss.grab(monitor)` already return an image cropped to the requested region (origin at `(0,0)`). Calling `_crop_screenshot()` afterwards applies absolute virtual-screen coordinates to this already-cropped image, which produces out-of-bounds/black output when the region doesn't start at the virtual screen origin.

Practical impact is very limited in the current codebase: the auto chain tries `dxcam` first (a required dependency), and the dxcam path does not call `_crop_screenshot`. The bug only manifests when dxcam is unavailable or fails and the capture rect targets a non-primary monitor — an unlikely combination in normal usage.

I'll verify and fix this in a follow-up PR to keep the refactoring scope clean.

File: src/windows_mcp/desktop/screenshot.py [hunk 2]
```diff
@@ -30,154 +53,209 @@ def get_screenshot_backend() -> str:
...
+                monitor = sct.monitors[0]
+            else:
+                monitor = {
+                    "left": capture_rect.left,
+                    "top": capture_rect.top,
+                    "width": capture_rect.right - capture_rect.left,
+                    "height": capture_rect.bottom - capture_rect.top,
+                }
+            raw = sct.grab(monitor)
+            image = Image.frombytes("RGB", raw.size, raw.rgb)
```
[the remark this answers is not in the document]
Comment (JezaChen) [reply src/windows_mcp/desktop/screenshot.py:207]: same as [above](https://github.com/CursorTouch/Windows-MCP/pull/202#discussion_r3067729412)

File: src/windows_mcp/desktop/screenshot.py [hunk 3]
```diff
@@ -30,154 +53,209 @@ def get_screenshot_backend() -> str:
...
+        return _crop_screenshot(image, capture_rect)
```
[the remark this answers is not in the document]
Comment (JezaChen) [reply src/windows_mcp/desktop/screenshot.py:208]: same as [above](https://github.com/CursorTouch/Windows-MCP/pull/202#discussion_r3067729412)
